src/helper.cc

```diff
@@ -1286,13 +1298,10 @@ GetFirstAvailable(const helper * hlp)
     for (n = hlp->servers.head; n != NULL; n = n->next) {
         srv = (helper_server *)n->data;
 
-        if (srv->canAcceptRequests())
-            continue;
-
         if (selected && selected->stats.pending <= srv->stats.pending)
             continue;
 
-        if (srv->flags.shutdown)
+        if (!srv->available())
```

Comment:
Please move this up, where canAcceptRequests() used to be. It feels wrong to compare `selected->stats.pending` and `srv->stats.pending` when `srv` is not actually available.